### PR TITLE
Fix for Springer reprocessing files

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: tests/units/common/test_sftp.py [hunk 1]
```diff
@@ -23,15 +22,7 @@ def initiate_sftp_service():
     raises(DirectoryNotFoundException, initiate_sftp_service)
 
 
-def test_logs():
-    with capture_logs() as cap_logs:
+def test_error_raise():
+    with raises(Exception):
```
Comment (ErnestaP) [inline tests/units/common/test_sftp.py:26]: Checking if the error is raised in __exit__.
It should be because we are returning False in __exit__:
https://github.com/cern-sis/workflows/blob/79606106d5e54b6d59f141ade34d2c885cd91303/dags/common/sftp_service.py#L55

File: dags/springer/parser.py [hunk 2]
```diff
@@ -115,6 +114,15 @@ def __init__(self) -> None:
...
 
+    def get_dois(self, article: ET.Element):
+        source = "./Journal/Volume/Issue/Article/ArticleInfo/ArticleDOI"
+        doi_element = article.find(source)
+        if doi_element is not None:
+            dois = doi_element.text
+            self.logger.msg("Parsing dois for article", dois=dois)
+            self.dois = dois
+            return [dois]
+
```
Comment (drjova) [inline dags/springer/parser.py]: this function will return only one doi, no?

File: tests/units/common/test_sftp.py [hunk 3]
```diff
@@ -23,15 +22,7 @@ def initiate_sftp_service():
     raises(DirectoryNotFoundException, initiate_sftp_service)
 
 
-def test_logs():
-    with capture_logs() as cap_logs:
+def test_error_raise():
```
Comment (drjova) [inline tests/units/common/test_sftp.py:25]: I don't understand why we need this?

File: dags/springer/parser.py (hunk 2 above)
Comment (ErnestaP) [reply dags/springer/parser.py]: Yes, we talked about this already before: because it is implemented to merge to doi arrays. It was made first in Springer parsing, before moving to enhancement. According to the schema, _dois_ field is array, wanted to change it before, but you said it's better to leave it for now

File: tests/units/common/test_sftp.py (hunk 3 above)
Comment (ErnestaP) [reply tests/units/common/test_sftp.py:25]: I tiried to explain it here: https://github.com/cern-sis/workflows/pull/92/files/76766090ae4f3fcc0a150d3a1644e49cc41acb99#r1100381724

I put the new return on condition, and wanted to test is it really throws an error.
If I woudn't return False in thi __exit__ function, it wouldn't throw an error, even if it was raised
docs: https://docs.python.org/3/reference/datamodel.html#object.__exit__

File: dags/springer/parser.py (hunk 2 above)
Comment (ErnestaP) [reply dags/springer/parser.py]: @drjova  should we open a new PR and change in all places that it infliences? It is not the first time this place raises questions.

File: dags/springer/dag_pull_ftp.py [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 @dag(
     start_date=airflow.utils.dates.days_ago(0),
-    schedule_interval="@hourly",
+    schedule_interval="30 */3 * * *",
```
Comment (drjova) [inline dags/springer/dag_pull_ftp.py:13]: why this change?

File: dags/springer/parser.py (hunk 2 above)
Comment (drjova) [reply dags/springer/parser.py]: No, it's ok for the moment thanks

File: tests/units/common/test_sftp.py (hunk 3 above)
Comment (drjova) [reply tests/units/common/test_sftp.py:25]: Apologies, but still I don't understand why we need this test. It catches any error extending `Exception`, meaning almost everything

File: dags/springer/parser.py (hunk 2 above)
Comment (drjova) [inline dags/springer/parser.py]:
```suggestion
        if doi_element is None:
            return []
        dois = doi_element.text
        self.logger.msg("Parsing dois for article", dois=dois)
        self.dois = dois
        return [dois]

```

File: dags/springer/dag_pull_ftp.py (hunk 4 above)
Comment (ErnestaP) [reply dags/springer/dag_pull_ftp.py:13]: because in the issue was mentioned:
Check and enable if is missing the schedule for its publisher, it should be the same as in PRODUCTION

File: tests/units/common/test_sftp.py (hunk 3 above)
Comment (ErnestaP) [reply tests/units/common/test_sftp.py:25]: I wanted to ensure that it catches whichever Exception and raises it. Before adding return False it was not raising anything, so the task was not stopped, even thought the connection so SFTP was not successful.

**Review by drjova — APPROVED** [review]
